### ISSUE TITLE
Unexpected 401 error despite successful login

### ISSUE THREAD

**Issue by hikkoiri** [inrupt/solid-client-authn-js#389]
**Describe the bug**
While creating a new Solid application, following the [Inrupt getting started documentation for javascript clients](https://docs.inrupt.com/developer-tools/javascript/client-libraries/tutorial/getting-started/), I was hit by an unexpected **HTTP 401 Unauthorized**, even though the client logged in successfully.

**To Reproduce**
```js
async function handleRedirectAfterLogin() {
  if (new URL(window.location.href).searchParams.get("code")) {
    await session.handleIncomingRedirect(window.location.href)
      .then(results => {

        console.log("Login succeeded:" + JSON.stringify(results))

        console.log(JSON.stringify(session.fetch))
        getSolidDataset("https://solid.carlo-hildebrandt.de/test", { fetch: session.fetch })
          .then(results => console.log(JSON.stringify(results)))
          .catch(err => console.log(JSON.stringify(err)))
      }
      )
      .catch(err => console.error(JSON.stringify(err)));
  }
}
```

I tried the same procedure with two different Identity - & Resource Provider. A self-hosted NSS and the Inrupt.net server.

**Expected behavior**
It is expected, that the code above fetches the private resource since the client is logged in and the resource exists.

**Screenshots**
The code above lead to this this error:

![image](https://user-images.githubusercontent.com/21198706/95213596-45873880-07ef-11eb-9b93-958d775065db.png)

The response message indicates, that the client is not authenticated:
![image](https://user-images.githubusercontent.com/21198706/95213794-7e271200-07ef-11eb-9a62-4b45c3a0f624.png)

**Desktop (please complete the following information):**

- OS: macOS (Catalina)
- Browser Chrome
- Version 85

**Additional context**
It is noticeable, that `session.fetch` is undefined. I would expect, that that should not be the case after a successful login.
Am I overlooking something obvious?  Already thanks for your support!

**Comment by NSeydoux** [comment]
Hello @hikkoiri ! We had a compatibility issue with NSS yesterday which may be causing the behaviour you are observing. The bug affects `@inrupt/solid-client-authn-browser@0.2.0`, and it has been patched in version `0.2.1`. What version of the library are you experiencing the bug with ?

Also, there's a known issue with redirection: for security reason, a request is authenticated for one given target only, which doesn't play nice with redirects. We're looking at how this could be worked around, but in the meantime, when requesting a container (or a folder if you prefer), the `/` must be present at the end of the URL, because if not the server redirects you to add it. It the screenshot you shared, you are trying to access `https://solid.carlo-hildebrandt.de/test/`, which is a container. If your fetch points to `https://solid.carlo-hildebrandt.de/test`, then you will get a 401.

Does any of these two suggestions help ?

**Comment by NSeydoux** [comment]
No problem, thanks for reporting the issue :)
